sycl/include/sycl/access/access.hpp

```diff
@@ -15,6 +15,16 @@
 #include <type_traits>
 #endif
 
+#ifdef __SYCL_DEVICE_ONLY__
```

Comment:
This doesn't look like a suitable place for this declaration. IMO, something under `sycl/__spirv` (or maybe even elsewhere) would be better.